internal/cache/file/cache_handle.go

```diff
@@ -94,35 +91,43 @@ func (fch *CacheHandle) shouldReadFromCache(jobStatus *downloader.JobStatus, req
 }
 
 // checkEntryInFileInfoCache checks if entry is present for a given object in
-// file info cache with same generation. It returns nil if entry is present,
-// otherwise returns an error with appropriate message.
-func (fch *CacheHandle) checkEntryInFileInfoCache(bucket gcs.Bucket, object *gcs.MinObject, requiredOffset int64) error {
+// file info cache with same generation and at least requiredOffset.
+// It returns nil if entry is present, otherwise returns an appropriate error.
+// Whether to change the order in cache while lookup is controlled via
+// changeCacheOrder.
+func (fch *CacheHandle) checkEntryInFileInfoCache(bucket gcs.Bucket, object *gcs.MinObject, requiredOffset uint64, changeCacheOrder bool) error {
 	fileInfoKey := data.FileInfoKey{
 		BucketName: bucket.Name(),
 		ObjectName: object.Name,
 	}
 	fileInfoKeyName, err := fileInfoKey.Key()
 	if err != nil {
-		return fmt.Errorf("read: while creating key: %v", fileInfoKeyName)
+		return fmt.Errorf("error while creating key for bucket %s and object %s: %w", bucket.Name(), object.Name, err)
 	}
-	// Look up of file being read in file info cache is required to update the LRU
-	// order on every read request from kernel i.e. with every read request from
-	// kernel, the file being read becomes most recently used.
-	fileInfo := fch.fileInfoCache.LookUp(fileInfoKeyName)
 
-	// The generation check below is required because it may happen that file
-	// being read is evicted from cache during or after reading the required offset
-	// from local cached file to `dst` buffer.
+	var fileInfo lru.ValueType
+	if changeCacheOrder {
+		fileInfo = fch.fileInfoCache.LookUp(fileInfoKeyName)
+	} else {
+		fileInfo = fch.fileInfoCache.LookUpWithoutChangingOrder(fileInfoKeyName)
```

Comment:
No action required, just a thought: our code has tightly coupled with LRU cache behavior, we might need to refactor, if we would support different cache policy.